pkg/controller/installation/core_controller.go

```diff
@@ -1394,6 +1394,18 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		return reconcile.Result{}, err
 	}
 
+	if imageSet == nil {
```

Comment:
I had to read this a few times. As a reader, I think I would prefer to have something like:

```
	required, err := imageset.IsRequired(ctx, r.client, imageSet, instance.Spec.Variant);
	if err != nil {
		r.status.SetDegraded(operator.ResourceReadError, "Error checking for non-variant ImageSet", err, reqLogger)
		return reconcile.Result{}, err
	} else if required {
		r.status.SetDegraded(...)
		return reconcile.Result{}, ...
	}

```

And inside of the imageset.IsRequired(..) explain what the requirements are. I think that would be the best place for those details.